spanner_orm/decorator.py

```diff
@@ -103,6 +104,12 @@ def wrapper(*args, **kwargs) -> T:
       return func(*args, **kwargs)
 
     spanner_api_method = spanner_api_method_lambda()
-    return spanner_api_method(spanner_wrapper, *args, **kwargs)
+    try:
+      return spanner_api_method(spanner_wrapper, *args, **kwargs)
+    except exceptions.NotFound:
```

Comment:
This is probably too generic to catch. There could be legitimate cases for NotFound, and I wouldn't want to indefinitely keep retrying just in case.
- Can you check the exception details to ensure it is session
- Can you also add some retry logic that eventually gives up
- Decorator usage is optional, we should probably add this somewhere deeper, maybe SpannerWriteApi and SpannerReadApi?

Reply:
To force a deletion, spanner has a session#delete api. I'm not sure if our python sdk has that but that could be an option for testing

Reply:
I moved the logic into SpannerWriteApi and SpannerReadApi so it works without the decorators. I also added a check for "session not found" to reduce false positives. I'd like to leave off the extra retry logic for now because the purpose of this change is not catch transient Spanner connection issues. It's designed to recover from rare connection issues and I think the code accomplishes that.

Reply:
I must have misread the original code, I thought it would keep retrying, more than once. This looks good